rootx/src/rootx.cxx

```diff
@@ -1,269 +0,0 @@
-#include <sys/user.h>
-#include <sys/types.h>
-#include <libutil.h>
-#include <libprocstat.h>
-#endif // R__FBSD
-
-#if defined(__CYGWIN__) && defined(__GNUC__)
-#define ROOTBINARY "root_exe.exe"
-#else
-#define ROOTBINARY "root.exe"
-#endif
-
-static int gChildpid;
-static int GetErrno()
-{
-   return errno;
-}
-
-static void ResetErrno()
-{
-   errno = 0;
-}
-
-static const char *GetExePath()
-{
-   static std::string exepath;
-   if (exepath == "") {
-#ifdef __APPLE__
-      exepath = _dyld_get_image_name(0);
-#endif
-#ifdef __linux
-      char linkname[64];      // /proc/<pid>/exe
-      char buf[kMAXPATHLEN];  // exe path name
-      pid_t pid;
-
-      // get our pid and build the name of the link in /proc
-      pid = getpid();
-      snprintf(linkname,64, "/proc/%i/exe", pid);
-      int ret = readlink(linkname, buf, kMAXPATHLEN);
-      if (ret > 0 && ret < kMAXPATHLEN) {
-         buf[ret] = 0;
-         exepath = buf;
-      }
-#endif
-#if defined(R__FBSD)
-  procstat* ps = procstat_open_sysctl();  //
-  kinfo_proc* kp = kinfo_getproc(getpid());
-
-  if (kp!=NULL) {
-     char path_str[PATH_MAX] = "";
-     procstat_getpathname(ps, kp, path_str, sizeof(path_str));
-     exepath = path_str;
-  }
-
-  free(kp);
-  procstat_close(ps);
-#endif
-   }
-   return exepath.c_str();
-}
-
-static void SetRootSys()
-{
-   const char *exepath = GetExePath();
-   if (exepath && *exepath) {
-      std::string epStr{exepath};
-      char *ep = epStr.data();
-      char *s;
-      if ((s = strrchr(ep, '/'))) {
-         *s = 0;
-         if ((s = strrchr(ep, '/'))) {
-            *s = 0;
-            int l2 = strlen(ep) + 10;
-            char *env = new char[l2];
-            snprintf(env, l2, "ROOTSYS=%s", ep);
-            putenv(env); // NOLINT: allocated memory now used by environment variable
-         }
-      }
-   }
-}
-
-extern "C" {
-   static void SigTerm(int);
-}
-
-static void SigTerm(int sig)
-{
-   // When we get terminated, terminate child, too.
-   kill(gChildpid, sig);
-}
-
-static void WaitChild()
-{
-   // Wait till child (i.e. ROOT) is finished.
-
-   int status;
-
-   do {
-      while (waitpid(gChildpid, &status, WUNTRACED) < 0) {
-         if (GetErrno() != EINTR)
-            break;
-         ResetErrno();
-      }
-
-      if (WIFEXITED(status))
-         exit(WEXITSTATUS(status));
-
-      if (WIFSIGNALED(status))
-         exit(WTERMSIG(status));
-
-      if (WIFSTOPPED(status)) {         // child got ctlr-Z
-         raise(SIGTSTP);                // stop also parent
-         kill(gChildpid, SIGCONT);       // if parent wakes up, wake up child
-      }
-   } while (WIFSTOPPED(status));
-
-   exit(0);
-}
-
-static void PrintUsage()
-{
-   fprintf(stderr, kCommandLineOptionsHelp);
-}
-
-int main(int argc, char **argv)
-{
-   char **argvv;
-   char  arg0[kMAXPATHLEN];
-
-#ifdef ROOTPREFIX
-   if (std::getenv("ROOTIGNOREPREFIX")) {
-#endif
-   // Try to set ROOTSYS depending on pathname of the executable
-   SetRootSys();
```

Comment:
There are still a few places in the code that looks up `ROOTSYS` (and thus could potentially lead to a difference in behavior between `root` and `root.exe`).  Some (eg the one in `TUnixSystem.cxx`) have additional search to find the 'right' value when not set) but (seemingly) not all. 

As part of this PR, we need to review those later ones and either remove if the result is not used or annotated them as needed.